crypto/s2n_rsa.c

```diff
@@ -98,8 +100,21 @@ static int s2n_rsa_encrypt(const struct s2n_pkey *pub, struct s2n_blob *in, stru
     S2N_ERROR_IF(out->size < size, S2N_ERR_NOMEM);
 
     const s2n_rsa_public_key *key = &pub->key.rsa_key;
-    int r = RSA_public_encrypt(in->size, ( unsigned char * )in->data, ( unsigned char * )out->data, key->rsa,
+#if S2N_GCC_VERSION_AT_LEAST(4,6,0)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
```

Comment:
These preprocess directives seems like they impact readability. Maybe we could confine them to one helper method that takes a const RSA* and returns a normal RSA*? It could have a scary name like "s2n_get_read_only_rsa"

Reply:
Also, how/why was that GCC version chosen? Is that the first version with "-Wcast-qual"?

Reply:
yep, agreed that we should put this all behind a macro. i punted on this because I think the current solution [wont work](https://github.com/aws/s2n-tls/pull/3459/commits/ac9753be8df28c0b0b40080b6d1f6e7cfcb34e04) and i am working on that first.

> Also, how/why was that GCC version chosen?
i got the 4.6 from a stackoverflow post but cant find any official docs stating the same so I'll need to revisit this.